### PR TITLE
add new blog Electronic_Magazine_Issue_4_hds_no2

### PR DESCRIPTION
**Pull request by kernelhe**
Hei,Mr liang,this is a new blog of Electronic_Magazine_Issue_4,please check!Thank you!

### CODE REVIEW

File: content/blog/2020/Electronic_Magazine_Issue_4_hds_no2/index.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+​		系统调用并非直接和程序员或系统管理员打交道，它仅仅是一个通过软中断机制（我们后面讲述）向内核提交请求，获取内核服务的接口。而在实际使用中程序员调用的多是用户编程接口——`API`，而管理员使用的则多是系统命令。
+
+​		用户编程接口其实是一个函数定义，说明了如何获得一个给定的服务，比如`read()、malloc()、free()、abs()`等。它有可能和系统调用形式上一致，比如`read()`接口就和`read`系统调用对应，但这种对应并非一一对应，往往会出现几种不同的API内部用到同一个系统调用，比如`malloc()、free()`内部利用`brk()`系统调用来扩大或缩小进程的堆；或一个API利用了好几个系统调用组合完成服务。更有些`API`甚至不需要任何系统调用——因为它并不是必需要使用内核服务，如计算整数绝对值的`abs()`接口。
+
+​		另外要补充的是Linux的用户编程接口遵循了在Unix世界中最流行的应用编程界面标准——`POSIX`标准，这套标准定义了一系列`API`。在Linux中（Unix也如此），这些`API`主要是通过C库（`libc`）实现的，它除了定义的一些标准的C函数外，一个很重要的任务就是提供了一套封装例程（`wrapper routine`）将系统调用在用户空间包装后供用户编程使用。
+
+​		不过封装并非必须的，如果你愿意直接调用，Linux内核也提供了一个`syscall()`函数来实现调用，我们看个例子来对比一下通过C库调用和直接调用的区别。
+
+```
```
Comment (helight) [inline content/blog/2020/Electronic_Magazine_Issue_4_hds_no2/index.md]:
```suggestion
```c
```

File: content/blog/2020/Electronic_Magazine_Issue_4_hds_no2/index.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+1. 编写名为`audit`的模块，其中除了加载、卸载模块函数以外，主要实现了`mod_sys_audit`与`mod_syscall_audit`两个函数。它们会分别挂载到`my_sysaudit`和`my_audit`两个钩子上。
+2. 编译后将模块加载 `insmod audit.o`。（你可通过`dmesg`查看加载信息）
+3. 修改`/usr/include/asm/unistd.h` ——在其中加入audit的系统调用号。这样用户空间才可以找到audit系统调用。
+4. 最后，我们写一个用户`deamon`程序，来循环调用audit系统调用，并把搜集到的信息打印到屏幕上。
+
+完了。系统调用还有许多细节，请大家查看有关书籍吧。不啰嗦了。再见。
+
+ 
+
+相关代码请下载  [auditexample.tar](http://wwww.kerneltravel.net/journal/iv/auditexample.tar.gz)（实现于2.4.18内核）。
```
Comment (ljrcore) [inline content/blog/2020/Electronic_Magazine_Issue_4_hds_no2/index.md]: 附件下载，可以放在博客对应的文件夹下

Comment (ljrcore) [reply content/blog/2020/Electronic_Magazine_Issue_4_hds_no2/index.md]: 请按上次周会时我提到的方法修改一下

**Comment by ljrcore** [comment]
good

**Pull request merged**